src/helpers/vyos-failover.py

```diff
@@ -91,32 +91,43 @@ def is_port_open(ip, port):
     finally:
         s.close()
 
-def is_target_alive(target=None, iface='', proto='icmp', port=None, debug=False):
+def is_target_alive(targets=None, iface='', proto='icmp', port=None, debug=False):
     """
     Host availability check by ICMP, ARP, TCP
-    Return True if target checks is successful
+    Return True if all target checks are successful
 
-    % is_target_alive('192.0.2.1', 'eth1', proto='arp')
+    % is_target_alive(['192.0.2.1', '192.0.2.5'], 'eth1', proto='arp')
     True
     """
     if iface != '':
         iface = f'-I {iface}'
-    if proto == 'icmp':
-        command = f'/usr/bin/ping -q {target} {iface} -n -c 2 -W 1'
-        rc, response = rc_cmd(command)
-        if debug: print(f'    [ CHECK-TARGET ]: [{command}] -- return-code [RC: {rc}]')
-        if rc == 0:
-            return True
-    elif proto == 'arp':
-        command = f'/usr/bin/arping -b -c 2 -f -w 1 -i 1 {iface} {target}'
-        rc, response = rc_cmd(command)
-        if debug: print(f'    [ CHECK-TARGET ]: [{command}] -- return-code [RC: {rc}]')
-        if rc == 0:
-            return True
-    elif proto == 'tcp' and port is not None:
-        return True if is_port_open(target, port) else False
-    else:
-        return False
+    is_targets_alive = []
+    for target in targets:
+        if proto == 'icmp':
+            command = f'/usr/bin/ping -q {target} {iface} -n -c 2 -W 1'
+            rc, response = rc_cmd(command)
+            if debug: print(f'    [ CHECK-TARGET ]: [{command}] -- return-code [RC: {rc}]')
+            if rc == 0:
```

Comment:
If all tests must pass, could all these `if rc == 0 / else` blocks instead be:
```
if rc != 0:
    return False
```
so it exits as soon as one fails, instead of always completing all tests?

Reply:
> If all tests must pass, could all these `if rc == 0 / else` blocks instead be:
> 
> ```
> if rc != 0:
>     return False
> ```
> 
> so it exits as soon as one fails, instead of always completing all tests?

Thanks! Fixed.